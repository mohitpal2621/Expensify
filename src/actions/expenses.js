import { v4 as uuid } from "uuid";

// ADD_EXPENSE action generator
export const addExpense = ({ description='', note='', amount=0, createdAt=0 }) => ({
    type: 'ADDEXPENSE',
    expense: {
        id: uuid(),
        description,
        note,
        amount,
        createdAt
    }
});

// REMOVE_EXPENSE action generator
export const removeExpense = (id) => ({
    type: 'REMOVEEXPENSE',
    id
});

// EDIT_EXPENSE action generator
export const editExpense = (id, updates) => ({
        type: 'EDITEXPENSE',
        id,
        updates
});