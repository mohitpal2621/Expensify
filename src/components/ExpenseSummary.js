import React from "react";
import { Link } from "react-router-dom";
import expensesTotal from '../selectors/expensesTotal';
import selectExpenses from '../selectors/expensesViewer';
import curr from "../format/currFormat";
import { useSelector } from "react-redux";

const ExpenseSummary = () => {
    const exp = useSelector(state => {
    
        return selectExpenses(state.expenses, state.filters)
    });
    const sum = expensesTotal(exp);
    return (
        <div className="page-header">
            <div className="content-container">
            {
                (exp.length >= 1) && <h1 className="page-header__title">Viewing <span>{exp.length}</span> {exp.length===1 ? "expense" : "expenses"} totalling <span>{curr.format(sum).replace("INR", "Rs.")}</span></h1>
            }
                <div className="page-header__actions">
                    <Link className="button hoverBtn" to="/create">Add Expense</Link>
                </div>
            </div>
        </div>
    );
};

export default ExpenseSummary;