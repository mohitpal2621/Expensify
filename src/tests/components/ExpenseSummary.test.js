import React from 'react';
import { render } from '@testing-library/react';
import ExpenseSummary from '../../components/ExpenseSummary';
import { useSelector } from 'react-redux';
import expenses from '../fixtures/expenses';

jest.mock('react-redux', () => ({
  ...jest.requireActual('react-redux'), // Use actual implementation of other hooks
  useSelector: jest.fn(), // Mock useSelector hook
}));

describe('ExpenseSummary component', () => {
  test('renders with single expense', () => {
    useSelector.mockReturnValue([expenses[1]]);

    const { container } = render(<ExpenseSummary />);

    expect(container).toMatchSnapshot();
  });

  test('renders with multiple expenses', () => {
    useSelector.mockReturnValue(expenses);

    const { container } = render(<ExpenseSummary />);

    expect(container).toMatchSnapshot();
  });
});
