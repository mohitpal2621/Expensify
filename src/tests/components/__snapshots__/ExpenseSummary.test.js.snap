// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExpenseSummary component renders with multiple expenses 1`] = `
<div>
  <div
    class="page-header"
  >
    <div
      class="content-container"
    >
      <h1
        class="page-header__title"
      >
        Viewing 
        <span>
          3
        </span>
         
        expenses
         totalling 
        <span>
          Rs. 127,362.74
        </span>
      </h1>
      <div
        class="page-header__actions"
      />
    </div>
  </div>
</div>
`;

exports[`ExpenseSummary component renders with single expense 1`] = `
<div>
  <div
    class="page-header"
  >
    <div
      class="content-container"
    >
      <h1
        class="page-header__title"
      >
        Viewing 
        <span>
          1
        </span>
         
        expense
         totalling 
        <span>
          Rs. 14,000.94
        </span>
      </h1>
      <div
        class="page-header__actions"
      />
    </div>
  </div>
</div>
`;
